### ISSUE TITLE
Unoptimal codegen for Vector<T> init if Vector is not a localVar.

### ISSUE THREAD

**Issue by sandreenko** [dotnet/runtime#33617]
For a method like this:
```
    struct StructWithVectorField
    {
        public int a;
        public Vector<float> b;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static void TestSIMDInit()
    {
        Vector<float> localVector = new Vector<float>();
        Console.WriteLine(localVector.ToString());
        StructWithVectorField structWithVectorField;
        structWithVectorField.a = 0;
        structWithVectorField.b = new Vector<float>();
        Console.WriteLine(structWithVectorField.b.ToString());
        Console.WriteLine(Vector<float>.Count);
    }
```
IL for these two Vector inits will look similar:

```
Vector<float> localVector = new Vector<float>();
->
IL_0000  12 00             ldloca.s     0x0
IL_0002  fe 15 01 00 00 1b initobj      0x1B000001 
```

```
 structWithVectorField.b = new Vector<float>();
->
IL_0024  7c 14 00 00 04    ldflda       0x4000014
IL_0029  fe 15 01 00 00 1b initobj      0x1B000001
```

but the generated code will be different:

```
1.
IN0001:        vxorps   ymm0, ymm0
IN0002:        vmovupd  ymmword ptr[V00 rsp+50H], ymm0
```

```
2.
IN0009:        vxorps   xmm0, xmm0
IN000a:        vmovdqu  xmmword ptr [V01+0x8 rsp+30H], xmm0
IN000b:        vmovdqu  xmmword ptr [V01+0x18 rsp+40H], xmm0
```

Of course, the first version is better (10 bytes against 16).

The problem happens because `fgMorphOneAsgBlockOp` doesn't do that optimization for `LCL_FLD` nodes, only for `LCL_VAR`.

[fieldUnoptimalVectorT.txt](https://github.com/dotnet/runtime/files/4336109/fieldUnoptimalVectorT.txt)

That will be probably fixed during https://github.com/dotnet/runtime/issues/1231.

Note: that is why we need to pass to `isBlkReqd == true` to  `fgMorphBlockOperand` for `gtType != TYP_STRUCT`.

The fix will cause diffs in ~10 pri1 tests like:
```
System.Linq.Set`1[Vector`1][System.Numerics.Vector`1[System.Single]]:Remove(System.Numerics.Vector`1[Single]):bool:this
System.Security.Principal.WindowsIdentity:RunImpersonated(Microsoft.Win32.SafeHandles.SafeAccessTokenHandle,System.Func`1[Vector`1]):System.Numerics.Vector`1[Single] (MethodHash=ffeb0926)
System.Threading.Tasks.RendezvousAwaitable`1[Vector`1][System.Numerics.Vector`1[System.Single]]:GetResult():System.Numerics.Vector`1[Single]:this (MethodHash=9b817fd3)
```
etc.

category:cq
theme:vector-codegen
skill-level:intermediate
cost:medium

**Comment by sandreenko** [comment]
cc @CarolEidt That is part of what we have discussed last week, I am still preparing the main issue description.

**Comment by AndyAyersMS** [comment]
@sandreenko is there work here beyond fixing #1231?

**Comment by sandreenko** [comment]
It is not part #1231, but after #1231 we will have more `LCL_VAR` and less `LCL_FLD` so the number of cases where we see this issue will be smaller.

But the real fix for that was in #33665, but I got a push back and was requested to do work that I do not have time to do currently.

**Comment by AndyAyersMS** [comment]
Ok, let's mark this as future.

**Comment by sandreenko** [comment]
Fixed by #55604
